feat: releases overview shows total docs in release

jordanl17 opened the pull request:
### Description
| Scenario | Before | After |
|--------|--------|--------|
| Active releases | <img width="1208" alt="Screenshot 2025-01-10 at 16 03 46" src="https://github.com/user-attachments/assets/76fdafd7-40ca-446d-b7fc-2694530de5a1" /> | <img width="1204" alt="Screenshot 2025-01-10 at 16 02 31" src="https://github.com/user-attachments/assets/99ad9f95-8cc6-4d75-ad60-ea33a9bcc6be" /> |
| Archived/Published releases | <img width="1206" alt="Screenshot 2025-01-10 at 16 04 20" src="https://github.com/user-attachments/assets/4145b3a0-3761-4b28-a215-54468c70be97" /> | <img width="1206" alt="Screenshot 2025-01-10 at 16 03 08" src="https://github.com/user-attachments/assets/9c89aa6e-8773-4da4-b7c3-f94257ce7b97" /> | 

* Split out the cell components for `ReleasesOverview` into separate components rather than all inside the column defs
* Added tests for these cell components
* Updated the cell component for the document count (`ReleaseDocumentsCount`), to:
  * use the `documentsMetadata` count for active releases
  * use the `finalDocumentStates` count for published or archived releases
* Removed `existingDocumentCount` from the release metadata aggregator store as it's no longer needed

<!--
What changes are introduced?
Why are these changes introduced?
What issue(s) does this solve? (with link, if possible)
-->

### What to review

<!--
What steps should the reviewer take in order to review?
What parts/flows of the application/packages/tooling is affected?
-->

### Testing
Added tests to the cell components
<!--
Did you add sufficient testing for this change?
If not, please explain how you tested this change and why it was not
possible/practical for writing an automated test
-->

### Notes for release
N/A
<!--
Engineers do not need to worry about the final copy,
but they must provide the docs team with enough context on:

* What changed
* How does one use it (code snippets, etc)
* Are there limitations we should be aware of

If this is PR is a partial implementation of a feature and is not enabled by default or if
this PR does not contain changes that needs mention in the release notes (tooling chores etc),
please call this out explicitly by writing "Part of feature X" or "Not required" in this section.
-->

jordanl17 commented on packages/sanity/src/core/releases/hooks/__tests__/useDocumentVersions.test.tsx:
This test change isn't entirely relevant to the task at hand, but I came across it and thought it would be good to try and move as quickly as possible to using the release fixtures everywhere

jordanl17 commented on packages/sanity/src/core/releases/store/createReleaseMetadataAggregator.ts:
Removing `existingDocumentsCount` from the store record

RitaDias commented on packages/sanity/src/core/releases/tool/overview/columnCells/ReleaseDocumentsCounter.tsx:
```diff
@@ -0,0 +1,31 @@
+    return (
+      <Translate
+        t={t}
+        i18nKey={documentCount > 1 ? `document-count_other` : `document-count_one`}
```
our i18n tool actually handles these causes automatically, as far as I know you shouldn't need to do this :) 
You need to send in the count like this (example on our code base, [translation with count](https://github.com/sanity-io/sanity/blob/sapp-1973/packages/sanity/src/core/comments/i18n/resources.ts#L74-L77), [use in component](https://github.com/sanity-io/sanity/blob/sapp-1973/packages/sanity/src/core/comments/plugin/field/CommentsFieldButton.tsx#L184))

jordanl17 replied:
Oh well that's quite nice! There are a few other places I've done this (the wrong) thing. I'll remember to go back and update them.

Thanks for flagging. However clever of them!

RitaDias reviewed (changes requested):
This looks great to me!
I guess the only thing would be about the translation key but otherwise everything looks good to go. My question is: are we going to add the separation of document changes (added, modified, deleted etc) at a later point or is that something that was dropped for now?

RitaDias commented:
Also, from a design stand point I really don't like how the numbers and text split the line, do we have to have the "documents" text there? :( Can't we add the header with the information that we are showing the count?
 
<img width="398" alt="image" src="https://github.com/user-attachments/assets/05e5c877-c814-4931-b4c4-a4fa3b2e0c74" />

jordanl17 commented:
> This looks great to me! I guess the only thing would be about the translation key but otherwise everything looks good to go. My question is: are we going to add the separation of document changes (added, modified, deleted etc) at a later point or is that something that was dropped for now?

That's been dropped now. The document action will only be present on release dashboard lists, but since this requires some backend work in places to make performant, we aren't going to touch it until a later date.
